src/daemon/cJsonParser.cc

```diff
@@ -65,32 +65,71 @@ bool cJsonParser::getConfig(struct sJsonDevicesConfig* pConfig)
         g_object_unref(pReader);
         return false; // failure
     }
-    auto node = json_reader_get_current_node(pReader);
-    if (json_node_get_node_type(node) != JSON_NODE_ARRAY) {
-        LOG_EVENT(LOG_ERR, "Error parsing 'devices': devices is not an array\n");
-        json_reader_end_member(pReader);
-        g_object_unref(pReader);
-        return false; // failure
-    }
-
-    auto devicePathArray = json_node_get_array(node);
-    if (devicePathArray == nullptr)
-    {
-        LOG_EVENT(LOG_ERR, "Error parsing 'devices': error getting array\n");
-        json_reader_end_member(pReader);
-        g_object_unref(pReader);
-        return false; // failure
-    }
 
-    json_array_foreach_element(devicePathArray, [](JsonArray *array,
-                                          guint index_, JsonNode *element_node,
-                                          gpointer user_data){
-            auto config = (struct sJsonDevicesConfig*) user_data;
-            auto devicePath = json_node_get_string(element_node);
+    if(JSON_CHECK_VERSION(1, 8, 0))
```

Comment:
It would be cleaner to have the API-specific routines encapsulated within helper functions, something like parseConfigPostV18 parseConfigPreV18 perhaps ? Or make the whole getConfig method being just an entry point for the version switch if it is even cleaner, your choice.

Reply:
Still doesn't build anyway:
```
| FAILED: CMakeFiles/KrillKounter.dir/src/daemon/cJsonParser.cc.o
| /build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot-native/usr/bin/x86_64-poky-linux/x86_64-poky-linux-g++ --sysroot=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot  -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot/usr/include/json-glib-1.0 -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot/usr/include/libmount -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot/usr/include/blkid -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot/usr/include/glib-2.0 -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot/usr/lib/glib-2.0/include -I/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/git/src -m64 -march=nehalem -mtune=generic -mfpmath=sse -msse4.2 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0=/usr/src/debug/krill-kounter/0.02-r0                      -fdebug-prefix-map=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0=/usr/src/debug/krill-kounter/0.02-r0                      -fdebug-prefix-map=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot=                      -fdebug-prefix-map=/build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden -Wshadow=local -std=gnu++20 -MD -MT CMakeFiles/KrillKounter.dir/src/daemon/cJsonParser.cc.o -MF CMakeFiles/KrillKounter.dir/src/daemon/cJsonParser.cc.o.d -o CMakeFiles/KrillKounter.dir/src/daemon/cJsonParser.cc.o -c /build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/git/src/daemon/cJsonParser.cc
| /build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/git/src/daemon/cJsonParser.cc: In member function 'bool cJsonParser::getConfig(sJsonDevicesConfig*)':
| /build/tmp/work/corei7-64-poky-linux/krill-kounter/0.02-r0/git/src/daemon/cJsonParser.cc:71:21: error: 'json_reader_get_current_node' was not declared in this scope; did you mean 'json_parser_get_current_pos'?
|    71 |         auto node = json_reader_get_current_node(pReader);
|       |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
|       |                     json_parser_get_current_pos
```

Reply:
ah ok, I'll just replace the old one; thought the linker might be happy since the code never runs

Reply:
compiler (not linker) needs a header definition

Reply:
oh I see what happened, accidentally used `if()` instead of `#if`